meeko/molsetup.py

```diff
@@ -4,623 +4,451 @@
+            )
+        return self.atoms[atom_index].is_ignore
 
-    @staticmethod
-    def get_bond_id(idx1, idx2):
-        """ used to generate canonical bond id from a pair of nodes in the graph"""
-        idx_min = min(idx1, idx2)
-        idx_max = max(idx1, idx2)
-        return (idx_min, idx_max)
+    def get_is_chiral(self, atom_index: int):
+        if atom_index > len(self.atoms) or self.atoms[atom_index].is_dummy:
+            raise IndexError(
+                "GET_IS_CHIRAL: provided atom index is out of range or is a dummy atom"
+            )
+        return self.atoms[atom_index].is_chiral
 
-    # replaced by
-    # def ring_atom_to_ring(self, arg):
-    #     return self.atom_to_ring_id[arg]
+    def get_neighbors(self, atom_index: int):
+        if atom_index > len(self.atoms) or self.atoms[atom_index].is_dummy:
+            raise IndexError(
+                "GET_GRAPH: provided atom index is out of range or is a dummy atom"
+            )
+        return self.atoms[atom_index].graph
+
+    def get_interaction_vectors(self, atom_index: int):
+        if atom_index > len(self.atoms) or self.atoms[atom_index].is_dummy:
+            raise IndexError(
+                "GET_INTERACTION_VECTORS: provided atom index is out of range or is a dummy atom"
+            )
+        return self.atoms[atom_index].interaction_vectors
+
+    # endregion
+
+    # def get_bonds_in_ring
+    # def walk_recursive
+    # def perceive_rings
+
+    # def merge_terminal_atoms(self, indices):
+    #     """
+    #     Primarily for merging hydrogens, but will merge the data for any atom or pseudoatom that is bonded to only one
+    #     other atom.
+    #
+    #     Parameters
+    #     ----------
+    #     indices: list
+    #         A list of indices to merge
+    #
+    #     Returns
+    #     -------
+    #     None
+    #     """
+    #     for index in indices:
+    #         if len(self.get_neighbors(index)) != 1:
+    #             msg = "Atempted to merge atom %d with %d neighbors. "
+    #             msg += "Only atoms with one neighbor can be merged."
+    #             msg = msg % (index + 1, self.get_neighbors(index))
+    #             raise RuntimeError(msg)
+    #         neighbor_index = self.get_neighbors(index)[0]
+    #         self.atoms[neighbor_index].charge += self.get_charge(index)
+    #         self.atoms[index].charge = 0.0
+    #         self.atoms[index].is_ignore = True
+    #     return
+
+
+# TODO: RENAME THIS TO NOT BE WORD VOMIT, CLEAN UP - Sets all the requirements for if you're building
+# A moleculesetup with an external toolkit like RDKit or OB
+class MoleculeSetupExternalToolBuild(ABC):
 
-    def get_bonds_in_ring(self, ring):
-        """ input: 'ring' (list of atom indices)
-            returns list of bonds in ring, each bond is a pair of atom indices
+    @staticmethod
+    def are_fourier_series_identical(fs1, fs2):
```

Comment:
Do you think this belongs to the molecule setup external tool base class or to the RDKit MoleculeSetup?

Reply:
I would put this in the base MoleculeSetup, it could, in theory, be used for a manually constructed (non-RDKit) molsetup.